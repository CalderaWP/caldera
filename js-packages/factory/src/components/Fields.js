import React, {Fragment} from 'react';
import {Field} from './Field';
import PropTypes from 'prop-types';
import classNames from 'classnames';
export const Fields = ({
						   fields, onChange, onBlur, className
					   }) => (
	<div className={classNames('caldera-fields-wrapper',className)}>
		{fields.forEach(field => <Field
			key={field.fieldId}
			{...field}
			onChange={onChange}
			onBlur={onBlur}/>)
		}
  </div>
);

Fields.propTypes = {
	field: PropTypes.shape({
		fieldId: PropTypes.string.isRequired,
		//input propTypes?
	}),
	onChange: PropTypes.func.isRequired,
	onBlur: PropTypes.func,
	className: PropTypes.string
};
