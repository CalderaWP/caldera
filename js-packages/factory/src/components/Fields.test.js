import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import {
	checkboxFieldset,
	selectField,
	checkboxField,
	numberField,
	textField,
	emailField
} from '../fields.fixtures';

import {Fields} from './Fields';

describe('Fields component', () => {
	let onChange;
	let onBlur;
	beforeEach(() => {
		onChange = jest.fn();
		onBlur = jest.fn()
	});
	it('Creates a text and checkbox field field', () => {
		const fields = [
			textField,
			checkboxField
		]
		const component = renderer.create(
			<Fields
				fields={fields} onChange={onChange} onBlur={onBlur}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Creates all five fields', () => {
		const fields = [
			checkboxFieldset,
			selectField,
			checkboxField,
			numberField,
			textField,
			emailField
		]
		const component = renderer.create(
			<Fields
				fields={fields} onChange={onChange} onBlur={onBlur}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

});
