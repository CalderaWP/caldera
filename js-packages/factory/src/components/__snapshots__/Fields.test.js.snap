// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields component Creates a text and checkbox field field 1`] = `
<div
  className="caldera-fields-wrapper"
/>
`;

exports[`Fields component Creates all five fields 1`] = `
<div
  className="caldera-fields-wrapper"
/>
`;
